frontend/model/notifications/mainNotificationsMixin.js

```diff
@@ -0,0 +1,156 @@
+'use strict'
+
+import sbp from '@sbp/sbp'
+import { PERIODIC_NOTIFICATION_TYPE } from './periodicNotifications/constants.js'
+import { compareISOTimestamps, comparePeriodStamps, DAYS_MILLIS, MONTHS_MILLIS } from '@model/contracts/shared/time.js'
+
+// util functions
+const myNotificationHas = checkFunc => {
+  return sbp('state/vuex/getters').currentNotifications.some(item => checkFunc(item))
+}
+
+/*
+ *** NOTE: This mixin is imported into the root Vue instance in 'main.js'. feel free to extend the notification entry lists
+           for defining more notifications
+
+  There is two types of notifications that can be defined here.
+  1) one-time notification: A type of notification that is checked & emitted once when user has signed in on their group.
+                            (Executed in response to 'SWITCH_GROUP' event)
+  2) periodic notification: A type of notication that is checked periodically and emitted when the conditions are met.
+
+  *-*-*-* notification entry structure *-*-*-*
+
+  1. oneTimeNotification: { name?: string, emitCondition: function, emit: function }
```

Comment:
@taoeffect 
_'one-time'_ notification here was not part of what you requested in the previous review, but while working on 'INCOME_DETAILS_OLD' notification, which is to check if the user's income details are >= 6 months old, I realised this one only needs to be checked 'once' when user has logs in on their group, not needing to be implemented as a periodic notification. So I worked on adding this too here and hope it sounds good.